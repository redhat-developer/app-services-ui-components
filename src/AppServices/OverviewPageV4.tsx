import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  Grid,
  Label,
  LabelGroup,
  Stack,
  StackItem,
  Title,
} from "@patternfly/react-core";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";
import type { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
import {
  AppServicesOverviewIconPattern,
  LogoRedHatOpenShiftApiManagementAStandardRgb,
  LogoRedHatOpenShiftConnectorsBStandardRgb,
  LogoRedHatOpenShiftDatabaseAccessAStandardRgb,
  LogoRedHatOpenShiftDataScienceAStandardRgb,
  LogoRedHatOpenShiftServiceRegistryAStandardRgb,
  LogoRedHatOpenShiftStreamsForApacheKafkaAStandardRgbPng,
  LogoRedHatOpenShiftAPIDesignerSvg,
} from "../images";
import {
  MarketingPageBanner,
  MarketingPageHero,
  MarketingPageSection,
} from "./components";

export type OverviewPageV4Props = {
  toKafkaHref: string;
  toServiceRegistryHref: string;
  toConnectorsHref: string;
  toAPIDesignHref: string;
};

export const OverviewPageV4: FunctionComponent<OverviewPageV4Props> = ({
  toKafkaHref,
  toServiceRegistryHref,
  toConnectorsHref,
  toAPIDesignHref,
}) => {
  const { t } = useTranslation();
  return (
    <>
      <MarketingPageHero
        title={t("overview-v4:heroTitle")}
        tagLine={t("overview-v4:heroTagline")}
        description={t("overview-v4:heroDescription")}
        description2={t("overview-v4:heroDescription2")}
        heroImage={AppServicesOverviewIconPattern}
        heroImageSize={678}
        cta={
          <Button
            variant={ButtonVariant.primary}
            ouiaId="button-try-kafka"
            isLarge
            component={(props) => (
              <Link
                {...props}
                data-testid="hero-buttonTryKafka"
                to={toKafkaHref}
              />
            )}
          >
            {t("overview-v4:heroCallToActionButton")}
          </Button>
        }
        variant="dark"
      />
      <MarketingPageBanner>{t("overview-v4:banner")}</MarketingPageBanner>
      <MarketingPageSection>
        <Grid md={6} lg={3} hasGutter>
          {/* Api Designer card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftAPIDesignerSvg}
                  alt="Red Hat OpenShift API Designer"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhoapidesignerTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="blue">
                      {t("overview-v4:applicationService")}
                    </Label>
                    <Label>{t("overview-v4:servicePreview")}</Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhoapidesignerMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhoapidesignerSecondaryText")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOAPIDesigner-buttonCTA"
                variant={ButtonVariant.secondary}
                component={(props) => (
                  <Link
                    {...props}
                    data-testid="hero-buttonCreateDesign"
                    to={toAPIDesignHref}
                  />
                )}
              >
                {t("overview-v4:rhoapidesignerCalltoActionButton")}
              </Button>
            </CardFooter>
          </Card>

          {/* API Management card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftApiManagementAStandardRgb}
                  alt="Red Hat OpenShift API Management logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhoamTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="blue">
                      {t("overview-v4:applicationService")}
                    </Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhoamMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhoamSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHOAM-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                variant={ButtonVariant.secondary}
                data-testid="cardRHOAM-buttonCTA"
                component="a"
                target="_blank"
                href="https://developers.redhat.com/products/rhoam/getting-started"
              >
                {t("overview-v4:getStarted")}{" "}
                <ExternalLinkAltIcon className="pf-u-ml-sm" />
              </Button>
            </CardFooter>
          </Card>

          {/* Kafka card */}
          <Card ouiaId="card-rhosak">
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftStreamsForApacheKafkaAStandardRgbPng}
                  alt="Red Hat OpenShift Streams for Apache Kafka logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhosakTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="blue">
                      {t("overview-v4:applicationService")}
                    </Label>
                    <Label color="green">{t("overview-v4:dataService")}</Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhosakMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhosakSecondaryText")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOSAK-buttonCreateKafka"
                variant={ButtonVariant.secondary}
                component={(props) => <Link {...props} to={toKafkaHref} />}
                ouiaId="button-create"
              >
                {t("overview-v4:rhosakCallToActionButton")}
              </Button>
            </CardFooter>
          </Card>

          {/* Service Registry card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftServiceRegistryAStandardRgb}
                  alt="Red Hat OpenShift Service Registry logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhosrTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="blue">
                      {t("overview-v4:applicationService")}
                    </Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhosrMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhosrSecondaryText")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOSR-buttonCTA"
                variant={ButtonVariant.secondary}
                component={(props) => (
                  <Link
                    {...props}
                    data-testid="hero-buttonTryKafka"
                    to={toServiceRegistryHref}
                  />
                )}
              >
                {t("overview-v4:rhosrCallToActionButton")}
              </Button>
            </CardFooter>
          </Card>

          {/* Connectors card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftConnectorsBStandardRgb}
                  alt="Red Hat OpenShift Connectors logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhocTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="blue">
                      {t("overview-v4:applicationService")}
                    </Label>
                    <Label>{t("overview-v4:servicePreview")}</Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhocMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhocSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHOC-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  {t("overview-v4:rhocSecondaryTextContinued")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOC-buttonCTA"
                variant={ButtonVariant.secondary}
                target="_blank"
                component={(props) => <Link {...props} to={toConnectorsHref} />}
              >
                {t("overview-v4:getStarted")}
              </Button>
            </CardFooter>
          </Card>

          {/* Database access card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftDatabaseAccessAStandardRgb}
                  alt="Red Hat OpenShift Data Science logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:dbaasTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="green">{t("overview:dataService")}</Label>
                    <Label>{t("overview-v4:servicePreview")}</Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:dbaasMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:dbaasSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHODS-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    {" "}
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHODS-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                target="_blank"
                href="https://console.redhat.com/application-services/databases"
              >
                {t("overview-v4:learnMore")}{" "}
                <ExternalLinkAltIcon className="pf-u-ml-sm" />
              </Button>
            </CardFooter>
          </Card>

          {/* Data science card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftDataScienceAStandardRgb}
                  alt="Red Hat OpenShift Data Science logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview-v4:rhodsTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <LabelGroup>
                    <Label color="green">{t("overview-v4:dataService")}</Label>
                    <Label>{t("overview-v4:fieldTrial")}</Label>
                  </LabelGroup>
                </StackItem>
                <StackItem>{t("overview-v4:rhodsMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview-v4:rhodsSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHODS-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    {" "}
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHODS-buttonTryIt"
                ouiaId="button-rhods-tryit"
                variant={ButtonVariant.secondary}
                component="a"
                target="_blank"
              >
                {t("overview-v4:tryIt")}
              </Button>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>
    </>
  );
};
