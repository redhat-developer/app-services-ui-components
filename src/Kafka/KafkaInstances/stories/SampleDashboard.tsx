export default () => <div>Dashboard</div>;
