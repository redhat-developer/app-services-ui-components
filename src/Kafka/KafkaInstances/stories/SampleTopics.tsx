export default () => <div>Topics</div>;
