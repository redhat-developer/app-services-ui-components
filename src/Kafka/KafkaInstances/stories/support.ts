import { addHours } from "date-fns";
import { useEffect, useState } from "react";
import type { KafkaInstance } from "../../types";

const now = new Date().toISOString();
const future = addHours(new Date(), 19).toISOString();
const INSTANCES: KafkaInstance[] = [
  {
    id: "1",
    name: "foo",
    createdAt: now,
    updatedAt: now,
    expiryDate: undefined,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "provisioning",
    plan: "standard",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "2",
    name: "bar",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "deleting",
    plan: "developer",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "3",
    name: "baz",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "azure",
    region: "region",
    status: "ready",
    plan: "standard",
    size: "2",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
];
export const useInstances = () => {
  const [instances, setInstances] = useState<KafkaInstance[]>();

  useEffect(() => {
    const timer = setTimeout(() => {
      setInstances(INSTANCES);
    }, 500);
    return () => clearTimeout(timer);
  }, []);
  return {
    instances,
    loading: instances === undefined,
  };
};
export const useInstance = (id: string) => {
  const { instances, loading } = useInstances();
  const instance = instances?.find((i) => i.id === id);

  return {
    instance,
    loading: loading || instance === undefined,
  };
};
