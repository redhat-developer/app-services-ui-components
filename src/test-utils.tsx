import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
