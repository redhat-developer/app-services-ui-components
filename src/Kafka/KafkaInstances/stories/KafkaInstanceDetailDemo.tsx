import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
