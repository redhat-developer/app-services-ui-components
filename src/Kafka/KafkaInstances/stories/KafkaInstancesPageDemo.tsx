import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
