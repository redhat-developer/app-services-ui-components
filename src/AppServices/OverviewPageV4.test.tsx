import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../test-utils";
import * as stories from "./OverviewPageV4.stories";

const { OverviewPageV4 } = composeStories(stories);

describe("OverviewPage", () => {
  it("renders", async () => {
    const comp = render(<OverviewPageV4 />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
    expect(
      comp.getByText("Try OpenShift Streams for Apache Kafka")
    ).toHaveAttribute("href", OverviewPageV4.args!.toKafkaHref);
    const [createKafka, createServiceRegistry] =
      comp.getAllByText("Create an instance");
    expect(createKafka).toHaveAttribute(
      "href",
      OverviewPageV4.args!.toKafkaHref
    );
    expect(createServiceRegistry).toHaveAttribute(
      "href",
      OverviewPageV4.args!.toServiceRegistryHref
    );
    const createConnectors = comp.getByTestId("cardRHOC-buttonCTA");
    expect(createConnectors).toHaveAttribute(
      "href",
      OverviewPageV4.args!.toConnectorsHref
    );

    const createDesign = comp.getByTestId("hero-buttonCreateDesign");
    expect(createDesign).toHaveAttribute(
      "href",
      OverviewPageV4.args!.toAPIDesignHref
    );

    expect(comp.getByTestId("cardRHODS-buttonTryIt")).toHaveAttribute(
      "data-ouia-component-id",
      "button-rhods-tryit"
    );

    expect(
      comp.getByText("Try OpenShift Streams for Apache Kafka")
    ).toHaveAttribute("data-ouia-component-id", "button-try-kafka");
  });
});
