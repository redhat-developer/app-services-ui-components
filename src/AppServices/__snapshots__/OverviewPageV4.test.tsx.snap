// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewPage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-dark-200 appsrv-marketing--banner pf-u-background-color-dark-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 678px; --appsrv-marketing--banner--before--BackgroundRepeat: repeat; --appsrv-marketing--banner--before--BackgroundPositionY: 0px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift Application and Data Services
        </h1>
        <p
          class="appsrv-marketing--banner__tagline "
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Cloud services for managed OpenShift
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Red Hat OpenShift Application and Data Services streamline the hybrid cloud experience, reducing the operational cost and complexity of delivering cloud-native applications.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-all-6-col-on-md pf-m-all-3-col-on-lg pf-m-gutter"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-1"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift API Management logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-2"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift API Management
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  API Management supports application development teams that want to take an API-first approach to building microservices-based applications.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is hosted and managed by Red Hat as an add-on for managed
                   
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardRHOAM-linkOpenShift"
                    href="https://openshift.com"
                    target="_blank"
                  >
                    OpenShift
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-xs"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                  .
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHOAM-buttonCTA"
                href="https://developers.redhat.com/products/rhoam/getting-started"
                target="_blank"
              >
                Get started
                 
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-sm"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift Data Science logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift Data Science
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  Data Science provides a fully-supported sandbox to rapidly develop, train, and test containerized machine learning models in the public cloud before deploying in production.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is hosted and managed by Red Hat as an add-on for managed
                   
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardRHODS-linkOpenShift"
                    href="https://openshift.com"
                    target="_blank"
                  >
                     
                    OpenShift
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-xs"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                  .
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="button-rhods-tryit"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHODS-buttonTryIt"
                target="_blank"
              >
                Try it
              </a>
            </div>
          </article>
        </div>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
